client/devdocs/docs-example/index.js

```diff
@@ -0,0 +1,89 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import history from 'lib/history';
+
+/**
+ * Internal dependencies
+ */
+import { getPath } from 'lib/nav-utils';
+import { Link } from '@woocommerce/components';
+
+// TODO Render README from this.props.readmeFilePath
+export default class DocsExample extends Component {
+	constructor() {
+		super();
+		this.state = {
+			currentPath: '/devdocs',
+		};
+	}
+
+	componentDidMount() {
+		/* eslint-disable */
+		this.setState( {
+			currentPath: getPath(),
+		} );
+		this.unlisten = history.listen( () => {
+			this.setState( {
+				currentPath: getPath(),
+			} );
+		} );
+		/* eslint-enable */
+	}
+
+	componentWillUnmount() {
+		this.unlisten();
+	}
+
+	// TODO Fix showing code example. Some components don't come through correctly (`<Gridicons />` becomes `<T />` for example
+	// and functions are output differently.
+	// TODO Add code highlighting.
+	// To use, add `jsx-to-string` to package.json
+	// <pre style={ { whiteSpace: 'pre-wrap' } }>{ this.renderCodeExample() }</pre>
```

Comment:
I was thinking we'd use something like what calypso's devdocs is doing, which actually goes in the reverse direction; [grabbing the code from a markdown file](https://github.com/Automattic/wp-calypso/blob/master/client/devdocs/gutenberg-components/example.jsx#L27-L37) & using [`gfm-code-blocks`](https://www.npmjs.com/package/gfm-code-blocks) to parse out the demo.

Reply:
🤦‍♂️ I was looking at what Calypso devdocs was doing for [Calypso components](https://github.com/Automattic/wp-calypso/blob/59bdfeeb97eda4266ad39410cb0a074d2c88dbc8/client/devdocs/design/playground-utils.js) but should have continued looking at the Gutenberg components.